src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private StorageDao storageDao;
+
+    public PurchaseOperationHandler() {
+        storageDao = new StorageDaoImpl();
```

Comment:
Consider using dependency injection to pass the `StorageDao` instance instead of instantiating it directly. This will improve testability and flexibility.